### ISSUE TITLE
Add support for non-selectable tree items

### ISSUE THREAD

**Issue by pcdavid** [eclipse-sirius/sirius-web#2123]
For the various modals we will need for the reference widget (see #2116, #2117, #2118) we will need tree representations which *display* all the elements in the project but only all the selection of the elements which are type-compatible with the reference being edited.

The frontend can not compute this type compatibility by itself, and we do not want to query it each time the user selects an element.

Instead we will add a new flag to `TreeItem`, named `selectable`.
It will be `true` by default so that the current behavior does not change.
When rendering a Tree, the backend will be able to set the flag individually for each item.

On the frontend, the non-selectable items will have a different style.
They will still be focusable/expandable/collapsable and have a context menu (if relevant for them).
The only behavioral difference is that they will not participate in the selection.
In other words, they will never invoke the provided `setSelection` callback.
